pkg/dyninst/gosym/symtab.go

```diff
@@ -627,6 +625,191 @@ func (lt *lineTable) locatePC(pc uint64, goFuncData []byte) []GoLocation {
 	return locations
 }
 
+func readInlTree(inlineTree []byte, fIdx uint32) (funcID uint8, nameOff uint32, parentPC uint32, ok bool) {
+	offset := inlinedCallSize * int(fIdx)
+	if offset+parentPCOffset+4 > len(inlineTree) {
+		return 0, 0, 0, false
+	}
+	funcID = inlineTree[offset+funcIDOffset]
+	nameOff = binary.LittleEndian.Uint32(inlineTree[offset+functionNameOffOffset:])
+	parentPC = binary.LittleEndian.Uint32(inlineTree[offset+parentPCOffset:])
+	ok = true
+	return
+}
+
+type fileRange struct {
+	pcLo uint64
+	pcHi uint64
+	file string
+}
+
+func (lt *lineTable) fileRanges(f *GoFunction) ([]fileRange, error) {
+	offset, found := f.funcInfo.pcfile()
+	if !found {
+		return nil, fmt.Errorf("no file data for function %s", f.Name)
+	}
+	offset += uint32(lt.pcTab[0])
+	if int(offset) >= len(lt.data) {
+		return nil, fmt.Errorf("file data offset out of range for function %s", f.Name)
+	}
+
+	names := map[uint32]string{}
+	pc := f.Entry
+	var ranges []fileRange
+	for file := range iterPCValues(lt.data[offset:], pc, lt.quantum) {
+		fno := uint32(file.Val)
+		var name string
+		var ok bool
+		if name, ok = names[fno]; !ok {
+			name = lt.fileName(f.funcInfo, fno)
+			names[fno] = name
+		}
+		ranges = append(ranges, fileRange{
+			pcLo: pc,
+			pcHi: file.PC,
+			file: name,
+		})
+		pc = file.PC
+	}
+	return ranges, nil
+}
+
+type functionRange struct {
+	pcLo uint64
+	pcHi uint64
+	name string
+	file string
+}
+
+func (lt *lineTable) inlinedFunctionsMapping(f *GoFunction, goFuncData []byte) []functionRange {
+	inlineTree := f.funcInfo.funcData(funcdataInlTree, goFuncData)
+	if inlineTree == nil {
+		return nil
+	}
+
+	offset, found := f.funcInfo.pcDataStart(pcdataInlTreeIndex)
+	if !found {
+		return nil
+	}
+	offset += uint32(lt.pcTab[0])
+	if int(offset) >= len(lt.data) {
+		return nil
+	}
+
+	files, err := lt.fileRanges(f)
+	if err != nil {
+		return nil
+	}
+	fileIdx := 0
+
+	names := map[uint32]string{}
```

Comment:
s/names/functionNames ?
There's a lot of names everywhere and multiple functions with similar structures; it's easy to get lost. Let's make the names explicit where we can.